pennylane/workflow/execution.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the :class:`ExecutionConfig` data class.
+"""
+from dataclasses import dataclass, field
+from typing import Callable, Union
+
+from pennylane.interfaces import SUPPORTED_INTERFACES
+
+
+@dataclass
+class ExecutionConfig:
+    """
+    A configuration class to describe an execution of a quantum circuit on a device.
+    """
+
+    shots: Union[None, int] = None
+    """The number of shots for an execution"""
+
+    gradient_method: Union[None, Callable] = None
```

Comment:
How do we want to represent backpropagation and device gradients here?

Reply:
this is the same as `diff_method` right? I'm good to have this copy the `diff_method` inputs if it makes more sense, but I don't love receiving strings when we can instead receive something like an enum, but those can be clumsy.

Reply:
Both `diff_method` provided to the QNode and `gradient_fn` provided to `execute` can be a string or a callable. Basically, the documentation for `qml.execute` is wrong, because `gradient_fn` can be either "device" or "backprop".